service-registry/registry-nacos/src/main/java/org/apache/servicecomb/registry/nacos/NacosDiscoveryInstance.java

```diff
@@ -57,7 +57,11 @@ public NacosDiscoveryInstance(Instance instance, String application, String serv
 
   @Override
   public MicroserviceInstanceStatus getStatus() {
-    return instance.isEnabled() ? MicroserviceInstanceStatus.UP : MicroserviceInstanceStatus.DOWN;
+    if (instance.isEnabled()){
+      return MicroserviceInstanceStatus.valueOf(instance.getMetadata().get(NacosConst.NACOS_STATUS));
```

Comment:
`instance.getMetadata().get(NacosConst.NACOS_STATUS` maybe null for old versions instance.